File: grid-pi.c
/* calculate Pi by computing whether every point in a grid is within the radius of a circle. */
#include <stdio.h>
#include <stdint.h>
#include <math.h>
#include <stdlib.h>
#include <inttypes.h>

int main (int argc, char **argv) {
	
	/* accumulators, final ratio is Pi */
	uint64_t acc_in_circle = 0;
	uint64_t acc_out_circle = 0;

	/* radius, such that 0,0 is origin and grid extents are +/- radius in x,y */
	int64_t radius = 40000;
	
	printf("Starting run, radius of %" PRIu64 "\n", radius);
	
	for (int64_t x = -radius; x <= radius; x++) {
		for (int64_t y = -radius; y <= radius; y++) {
			/* if Pythagoras says this point is outside the radius */
			if (((x * x) + (y * y)) > (radius * radius)) {
				acc_out_circle++;
			} else {
				acc_in_circle++;
			}
		}
	}
	
	long double proportion = 4 * ((long double) acc_in_circle / (acc_out_circle + acc_in_circle));
	
	printf("Run completed.\n");
	printf("Total points: %" PRIu64 "\n", (acc_in_circle + acc_out_circle));
	printf("Inside of circle: %" PRIu64 "\n", acc_in_circle);
	printf("Outside of circle: %" PRIu64 "\n", acc_out_circle);
	printf("Pi seems to be... %1.10Lf\n", proportion);
	
	return 0;
}

File: grid-pi.swift
// var acc_in_circle, acc_out_circle: Double;
var acc_in_circle, acc_out_circle: UInt;
acc_in_circle = 0;
acc_out_circle = 0;

let radius = 4000;

print("Starting run, radius of \(radius)");

for x in -radius...radius {
    for y in -radius...radius {
        if ((x * x) + (y * y)) > (radius * radius) {
            acc_out_circle += 1;
        } else {
            acc_in_circle += 1;
        }
    }
}

var proportion: Double;
proportion = 4 * (Double(acc_in_circle) / Double((acc_out_circle + acc_in_circle)));

print("Run completed.");
print("Total points: \(acc_in_circle + acc_out_circle)")
print("Inside of circle: \(acc_in_circle)")
print("Outside of circle: \(acc_out_circle)")
print("Pi seems to be... \(proportion)")


/*
# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
*/
